ledger/block/src/transition/mod.rs

```diff
@@ -309,6 +309,44 @@ impl<N: Network> Transition<N> {
         self.program_id.to_string() == "credits.aleo" && self.function_name.to_string() == "unbond"
     }
 
+    /// Returns `true` if this is a `bond_public` transition.
```

Comment:
Frankly speaking I have no idea why this function is marked as inline because there are so much code inside of them including at least 4 memory allocations.